app/models/maestrano/connector/rails/services/data_consolidator.rb

```diff
@@ -81,6 +81,9 @@ def consolidate_external_entities(external_entities, connec_entity_name)
         entity_id = @current_entity.class.id_from_external_entity_hash(entity)
         idmap = @current_entity.class.find_or_create_idmap(external_id: entity_id, organization_id: @organization.id, connec_entity: connec_entity_name.downcase)
 
+        # Don't push to connec, the immutable entities which have already been pushed
+        next if immutable_and_already_pushed_to_connec?(idmap)
```

Comment:
I personally think this is already baked into behavior (see comments).

If we are convinced we need the explicitness of this immutability, then I would prefer to do: 

``` 
next if immutable_and_already_pushed_to_connec?(idmap)
```

So we have the option to send immutable entities to quickbooks anyway in case of data integrity issues.